sqlglot/optimizer/scope.py

```diff
@@ -274,6 +278,7 @@ def columns(self):
                     not ancestor
                     or column.table
                     or isinstance(ancestor, exp.Select)
+                    or (isinstance(ancestor, exp.Table) and not isinstance(ancestor.this, exp.Func))
```

Comment:
The check for `Func` here was necessary because the following test was failing without it:

```sql
-- `hive_partition` is qualified without the check
SELECT x FROM READ_PARQUET('path.parquet', hive_partition=1);
```